modules/_labs/tabs/react/lib/Tab.tsx

```diff
@@ -120,11 +124,19 @@ const StyledButton = styled('button')<{isSelected: boolean} & StyledType>(
 
 export const Tab = createComponent('button')({
   displayName: 'Tabs.Item',
-  Component: ({name = '', model, children, ...elemProps}: TabProps, ref, Element) => {
+  Component: ({name = '', index, model, children, ...elemProps}: TabProps, ref, Element) => {
     const {localRef, elementRef} = useLocalRef(ref);
     const {state, events} = useModelContext(TabsModelContext, model);
 
-    const tabName = useRegisterItem({state, events}, localRef, name);
+    const tabName = useRegisterItem({state, events}, localRef, name, index);
+
+    if (
+      index &&
+      state.items.length &&
+      !state.items.find((item, i) => item.index === i && item.index === index && item.id === name)
+    ) {
+      events.updateItemPosition({id: name, index});
+    }
```

Comment:
`events` shouldn't be called directly in a render function since `events` typically call `React.setState` internally. This is also common functionality for any component using a list, so we should make a hook out of this. This would be a `ListModel` behavior, so it goes with `useRegisterItem`. Or perhaps it is rolled into `useRegisterItem` (since you pass the index anyways).

I'm also unsure if the model needs to record the index at all. It isn't really a concern of the model and the `state.items.find` can be expensive per render.

My suggestion is to move this logic into the `useRegisterItem` hook and add a `useLayoutEffect` section that tracks changes to the `index` and when the index changes, call `events.updateItemPosition`. It might look like this:

```tsx
export const useRegisterItem = (
  {state, events}: ListModel,
  localRef: React.RefObject<any>,
  id = '',
  index?: number
): string => {
  const [localId, setLocalId] = React.useState(id);
  const firstRender = React.useRef(true);

  useMountLayout(() => {
    const defaultId = state.indexRef.current;
    const itemId = id || String(defaultId);
    events.registerItem({item: {id: itemId, ref: localRef, index}});
    setLocalId(itemId);
    firstRender.current = false;

    return () => {
      events.unregisterItem({id: itemId});
    };
  });

  useLayoutEffect(() => {
    // Don't update on first render
    if (firstRender.current) {
      return;
    }
    
    events.updateItemPosition({id: name, index})

  }, [index])

  return localId;
};
```